dashboard/app/dsl/script_dsl.rb

```diff
@@ -286,14 +284,9 @@ def i18n_hash
     i18n_lesson_group_strings = {}
     @lesson_groups.each do |lesson_group|
       if lesson_group[:key]
-        i18n_lesson_group_strings[lesson_group[:key]] = {
-          'display_name' => lesson_group[:display_name]
-        }
-        if lesson_group[:description]
-          i18n_lesson_group_strings[lesson_group[:key]][:description] = lesson_group[:description]
-        end
-        unless lesson_group[:big_questions].nil_or_empty?
-          i18n_lesson_group_strings[lesson_group[:key]][:big_questions] = lesson_group[:big_questions]
+        i18n_lesson_group_strings[lesson_group[:key]] = {}
+        lesson_group.select {|k, _| [:display_name, :description, :big_questions].include?(k)}.each do |k, v|
```

Comment:
Could this be simplified even more by directly assigning the entire result of `select`? Something like:

```
i18n_lesson_group_strings[lesson_group[:key]] = lesson_group.select ...
```

Also, could the `nil_or_empty` check go into the condition inside the select block?

Reply:
I moved the nil or empty check but I'm not sure how to make the other change work. I'm going to merge it this way and can always follow up.

Reply:
It's completely fine as is, but if you were curious, I was thinking something like this:

```
        keys = [:display_name, :description, :big_questions]
        strings = lesson_group.select {|k, v| keys.include?(k) && !v.nil_or_empty?}.stringify_keys
        i18n_lesson_group_strings[lesson_group[:key]] = strings
```

(Lines 2 and 3 on separate lines purely for readability / line length reasons)

YMMV on whether that's better, though. I generally prefer doing things more "functionally" if it's easy to do. The nice thing here is if I just read `strings = lesson_group.select ...`, even without seeing the rest of the block, I know that `strings` is just going to be a subset of the `lesson_group` hash. Whereas anything can happen in a for loop.

Reply:
TIL `stringify_keys` thats definitely a helpful thing. Thanks for sharing this winter!